### DISCUSSION TITLE
Using certs from the filesystem rather than a secret on Kubernetes

### DISCUSSION THREAD

**Discussion by Sam-Kramer** [minio/minio#12715] [Q&A]
Hi, 

Our K8s infrastructure is a bit opinionated, in that we mount certs via "the filesystem" (volume) rather than secrets. How would you suggest, we specify the external cert (& ca) for tenants? Thankfully the certs are all in PEM format, but need to move directories/be renamed.

Reading through the code, it looks like as long as auto cert is disabled, no external cert is set, and we do not have KES enabled, we _could_ mount to the cert directory ourselves: https://github.com/minio/operator/blob/88f721b721a681e1112acd72c24f6bc954823b99/pkg/resources/statefulsets/minio-statefulset.go#L198-L203

Only problem then, is we need to munge the cert directory (names & structure) to match what Minio expects. 

Thanks,
Sam

**Comment by harshavardhana** [comment]
// cc @Alevsk

**Reply by Sam-Kramer** [reply]
To provide a bit of an update on this: I tried doing as I mentioned, which works and doesn't work. As there are no external certs provided, minio sets the env variable urls to be `http` vs `https`, and on start up we fail due to having certs provided, but tls being disabled (https://github.com/minio/operator/blob/f58fb62346f6c7844e85475f222131ef3daf1ccd/pkg/apis/minio.min.io/v2/helper.go#L404 & https://github.com/minio/minio/blob/d00783c92388f03eeac3f59fee5c955a22702fc3/cmd/server-main.go#L475-L477). 

However, one silver-lining I can potentially see here is if initContainers are exposed, we could read and then _update_ the secret, and wait until the secret is refreshed on disk (for the base-case, we can either create a dummy secret, or simply launch without tls until our secret has been created by the initContainer)

**Reply by Sam-Kramer** [reply]
Also, this seems to be not an uncommon pattern in the wild to have secrets be stored on the fs in a container (rather than managed via secrets). See: https://learn.hashicorp.com/tutorials/vault/kubernetes-sidecar
